File: backend/src/main/java/com/gpuoj/controller/AdminController.java
package com.gpuoj.controller;

import com.gpuoj.service.ProblemService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/admin")
@RequiredArgsConstructor
public class AdminController {

    private final ProblemService problemService;

    @PostMapping("/problems/sync")
    public Mono<String> syncProblems() {
        return problemService.syncProblems();
    }
}

File: backend/src/main/kotlin/com/gpuoj/controller/AdminController.kt
package com.gpuoj.controller

import com.gpuoj.service.ProblemService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/admin")
class AdminController(private val problemService: ProblemService) {

    @PostMapping("/problems/sync")
    fun syncProblems(): Mono<String> = problemService.syncProblems()
}
